### DISCUSSION TITLE
Pack fails with NETSDK1194 on .NET SDK 7.0.200

### DISCUSSION THREAD

**Discussion by JamieMagee** [cake-build/cake#4121] [General]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

3.0.0

### Operating system

Linux, N/A

### Operating system architecture

64-Bit

### CI Server

GitHub Actions

### What are you seeing?

When using `DotNetPack` and setting `OutputDirectory` I get:

```
MSBuild version 17.5.0-preview-23061-01+040e2a90e for .NET
Error: /usr/share/dotnet/sdk/7.0.200/Current/SolutionFile/ImportAfter/Microsoft.NET.Sdk.Solution.targets(36,5): error NETSDK1194: The "--output" option isn't supported when building a solution.
````

I think this was accidentally included as a breaking change, but was intended to be in .NET 8.
There is a workaround/fix listed here: https://github.com/dotnet/sdk/issues/30624#issuecomment-1432118204

### What is expected?

The build successfully completes

### Steps to Reproduce

Create a cake configuration with DotNetPack and an `OutputDirectory`.

**Comment by pascalberger** [comment]
Not sure that fixing breaking changes in underlying tools is a route Cake should go. The aliases are currently thin wrappers around the tools (in this case MsBuild), without containing much logic.

If we start adding logic to fix bugs, like in this case, in the underlying tools, we also take the burden of maintaining the fix in subsequent versions.

Also in this case Microsoft stopped rolling out Docker images of 7.0.200 and wait until next service release, where they will adopt it, making 7.0.200 likely be the only affected version.

**Reply by JamieMagee** [reply]
I understand not wanting to fix this for .NET 7, but I think Cake will need to make changes once .NET 8 rolls out and `--output` is actually removed.

**Reply by devlead** [reply]
> I understand not wanting to fix this for .NET 7, but I think Cake will need to make changes once .NET 8 rolls out and `--output` is actually removed.

Is it actually removed for project files too? Thought it only was for solution files.
